### ISSUE TITLE
UDP host port mapping is not working in v1.8 cluster created with kops

### ISSUE THREAD

**Issue by activeshadow** [projectcalico/kube-controllers#212]
## Expected Behavior
Given the below pod port configuration, I would have expected the k8s node running the pod to be listening on port `udp:5501`.
```
        ports:
        - containerPort: 5501
          hostPort: 5501
          protocol: UDP
```

## Current Behavior
The k8s node running the pod is **not** listening on `udp:5501`.

## Context
Since AWS doesn't support UDP load balancing, I'm trying to create my own poor man's load balancer using a weighted round robin DNS configuration. I cannot use a `NodePort` service because I need the port clients connect to to be `5501` for legacy reasons. So, I'm trying to use host port mapping, along with node selectors, do get this working.

I've ensured that the `portmap` executable is present on the nodes in `/opt/cni/bin/`, and I've checked the `kubelet` logs to make sure I didn't see any errors regarding the execution of `portmap`. I've also ensured that `/etc/cni/net.d/10-calico.conflist` is present and contains the `portmap` plugin configuration.

## Your Environment

* Kubernetes version (`kubectl version`): `v1.8.6`
* Calico version: `quay.io/calico/cni:v1.11.0`, `quay.io/calico/node:v2.6.2`
* Manifest file used to deploy k8s policy: I *think* the relevant portion is above
* Operating System and version: `Debian 8.10 (jessie)` (via kops)
* Link to your project (optional):

**Comment by activeshadow** [comment]
Seems like this is related to #109, but I'm not sure if using UDP makes a difference.

**Comment by caseydavenport** [comment]
> The k8s node running the pod is not listening on udp:5501.

Have you checked to see if the service is actually available on udp:5501?

The hostPort mapping is done via NAT in iptables, which means you won't see any process bound to that port, but it should still work.

**Comment by activeshadow** [comment]
Well... looks like I’m an idiot. I failed to realize it was being done via iptables (which I should have known). I checked and the rules are getting created and are pointing to the correct pod IP.

At this point it looks like I need to ensure 1) traffic is making it into the pod, and if so 2) why my application isn’t seeing it.

But... I don’t think it’s a Calico CNI problem so I’m gonna close this issue.

Thanks!
